### ISSUE TITLE
Click.on(invalid Target) produces misleading error

### ISSUE THREAD

**Issue by SiKing** [serenity-bdd/serenity-core#2138]
I had accidentally defined an invalid `Target`. Something like:
```
public static Target FIELD_TYPE_CANCEL = Target.the("Field Type selection cancel")
    .located(By.xpath("//div[contains(@class, 'no-bracket']/i[text()='cancel']"));
```
The above XPath is missing the closing bracket for `contains()`.

When I tried to click on this with:
```
when(chuck).attemptsTo(Click.on(FIELD_TYPE_CANCEL));
```
I got back:
```
net.thucydides.core.webdriver.exceptions.ElementShouldBeEnabledException: Expected enabled element was not enabled
[ stacktrace omited ]
```

This error is misleading.

If I try something like:
```
FIELD_TYPE_CANCEL.resolveFor(chuck).getAttribute("outerHTML");
```
I get back a much more correct and helpful:
```
org.openqa.selenium.InvalidSelectorException: invalid selector: Unable to locate an element with the xpath expression //div[contains(@class, 'technical-custom-field-type-filter-customFields']/i[text()='cancel'] because of the following error:
SyntaxError: Failed to execute 'evaluate' on 'Document': The string '//div[contains(@class, 'technical-custom-field-type-filter-customFields']/i[text()='cancel']' is not a valid XPath expression.
[ stacktrace omited ]
```

**Comment by SiKing** [comment]
I tried to trace where things _could_ have gone wrong. This is my best **guess**.

That error is thrown in only two places:
```
$ find serenity-core -type f -exec grep -Hn "Expected enabled element was not enabled" {} \;
serenity-core/src/main/java/net/serenitybdd/core/pages/WebElementFacadeImpl.java:1158:            throw new ElementShouldBeEnabledException("Expected enabled element was not enabled", timeout);
serenity-core/src/main/java/net/serenitybdd/core/pages/WebElementFacadeImpl.java:1170:            throw new ElementShouldBeEnabledException("Expected enabled element was not enabled", timeout);
```

[waitUntilClickable()](https://github.com/serenity-bdd/serenity-core/blob/master/serenity-core/src/main/java/net/serenitybdd/core/pages/WebElementFacadeImpl.java#L1157) calls [waitForCondition()](https://github.com/serenity-bdd/serenity-core/blob/34c788ed60dc14320c1c8e8507c703debac7a34b/serenity-core/src/main/java/net/serenitybdd/core/pages/WebElementFacadeImpl.java#L1076) which has `.ignoring(NoSuchElementException.class ...`
The "desirable" `InvalidSelectorException` extends `NoSuchElementException`.
Could this be the source of this issue?

**Comment by wakaleo** [comment]
You could be onto something. Do you want to modify the code to cater for that exception explicitly and return a more meaningful exception? If you are waiting on an element, it would be logical to fail immediately if the locator is invalid.

**Comment by SiKing** [comment]
I am not sure where would be the best place to do this.
Try to access this element in waitUntilClickable(), before calling waitForCondition(), and look for this exception?
Or modify waitForCondition() to not ignore specifically this exception, but ignore all the others - I think there are half-dozen that extend NoSuchElementException, so ignore only the others?

**Comment by wakaleo** [comment]
I think modifying `waitForCondition()` would be the simplest approach. But it would be tricky because you would need to ignore `InvalidSelectorException` but not it's parent class `NoSuchElementException`.

**Comment by SiKing** [comment]
This fix was released in 2.2.12.
